packages/slice-machine/components/Simulator/components/Header/index.tsx

```diff
@@ -39,6 +55,12 @@ const Header: React.FunctionComponent<PropTypes> = ({
   handleScreenSizeChange,
   size,
 }) => {
+  const { generateSliceScreenshot } = useSliceMachineActions();
+  const [data, setData] = useState<ScreenShotSyncState>(initialState);
+  const onTakingSliceScreenshot = () => {
+    generateSliceScreenshot(variation?.id, Model, setData);
```

Comment:
This is wrapped in a checkSimulatorSetup elsewhere. What's the purpose for it and is it useful here at all?

Reply:
Hey, in order to be able to take screenshots in the context of the user's project running, we need him to install a new page in his project called slice-simulator. (to render independent components with the user style).

This checkSimulatorSetup is a helper to always verify the user has every setup so that we can take a screenshot.
It is very useful 😄